subsys/ipc/ipc_service/backends/ipc_rpmsg_static_vrings.h

```diff
@@ -163,5 +163,5 @@ static inline unsigned int optimal_num_desc(size_t mem_size, unsigned int buf_si
 		num_desc++;
 	}
 
-	return (1 << (find_msb_set(num_desc) - 1));
+	return (1 << (find_msb_set(--num_desc) - 1));
```

Comment:
what's happen if num_desc == 0 ?

Reply:
> what's happen if num_desc == 0 ?

not possible, as it is initialized to 1...

Reply:
yeah, I'll add a check in a follow up PR.

Reply:
> not possible, as it is initialized to 1...

Yeah, but there is still a problem here. If the available space is not enough `num_desc == 1` and `--num_desc == 0`, so the function is still returning  `(1 << (find_msb_set(0) - 1))` that is an absurd value.